proxy/tests/support/proxy.rs

```diff
@@ -130,35 +130,49 @@ fn run(proxy: Proxy, mut env: config::TestEnv) -> Listening {
         config.metrics_flush_interval = dur;
     }
 
-    let mock_orig_dst = MockOriginalDst(Arc::new(Mutex::new(mock_orig_dst)));
```

Comment:
All this is just moving the creation of a `Main` into the thread, since `Main` is no longer `Send` (it owns a `Core`).